### ISSUE TITLE
App is inconsistent in whether it resets the  LocalPlayer after downloading an episode that is being streamed

### ISSUE THREAD

**Issue by mchowning** [Automattic/pocket-casts-android#1509]
### Description

I've noticed that when downloading a playing episode (i.e., if you start playing a new episode and you have auto download episodes added to up next turned on), once the download is complete, we usually, but not always reset the player.

I have only briefly looked into the issue, and I don't have time to look into it further, so I'm filing this issue as a reminder to follow-up on this when we have more time. At this point, it is not clear to me why we reset the player sometimes and not others.

It seems like either:
1. We should be resetting the player, and perhaps there are some negative consequences when we don't reset the player (although it's not clear to me right now what those are). I thought that maybe we were continuing to stream the podcast in that scenario, but that doesn't seem to be the case.
2. We do not need to be resetting the player every time we download and we can avoid doing that, which would be nice because it would let us avoid having the audio pause for approximately 1 second after the download completes while the player is being reset, or
3. There's a good reason we only need to reset the player some of the time, and I just haven't identified yet.

### Step-by-step reproduction instructions

1. Turn on the setting to auto download podcasts added to up next
2. Start playing a podcast that hasn't been downloaded
3. Observe that once the download completes, often, the logs show that we're resetting the player and there is an ~1 second pause in the audio. But observe that this does not always happen.

### Did you search for existing bug reports?

- [X] I have searched for existing bug reports.

### Device, Operating system, and Pocket Casts app version

* Samsung A13, Android 13, Pocket Casts pre-release 7.52 build
* Pixel 4 emulator, Android 13, Pocket Casts pre-release 7.52 build

**Comment by mchowning** [comment]
In my testing, it _seems_ that we either reset or do not reset the player repeatedly. In other words, if we reset the player on the first podcast you play, then it is more likely that we will keep resetting the player on subsequent podcasts you play. Likewise, if the app does not reset the player for a given podcast, it seems that will usually lead to subsequent podcasts not resetting the player as well. 

Basically, it does not appear that whether or not a given podcast resets the player when the download completes is independent of whether we reset the player on previous podcasts. Or, to put it another way, it seems like a given behavior tends to happen repeatedly, until it doesn't, and then the opposite behavior will happen multiple times in a row.
